lib/ruby_lsp/requests/support/rubocop_diagnostic.rb

```diff
@@ -102,6 +113,64 @@ def offense_replacements
             )
           end
         end
+
+        sig { returns(Interface::CodeAction) }
+        def disable_line_action
+          Interface::CodeAction.new(
+            title: "Disable #{@offense.cop_name} for this line",
+            kind: Constant::CodeActionKind::QUICK_FIX,
+            edit: Interface::WorkspaceEdit.new(
+              document_changes: [
+                Interface::TextDocumentEdit.new(
+                  text_document: Interface::OptionalVersionedTextDocumentIdentifier.new(
+                    uri: @uri.to_s,
+                    version: nil,
+                  ),
+                  edits: line_disable_comment,
+                ),
+              ],
+            ),
+          )
+        end
+
+        sig { returns(T::Array[Interface::TextEdit]) }
+        def line_disable_comment
+          new_text = if @offense.source_line.include?(" # rubocop:disable ")
+            ",#{@offense.cop_name}"
+          else
+            " # rubocop:disable #{@offense.cop_name}"
+          end
+
+          eol = Interface::Position.new(
+            line: @offense.line - 1,
+            character: length_of_line(@offense.source_line),
+          )
+
+          # TODO: probably better to use a block comment for multi-line
+          # offenses. Harder though; need to do indentation and such.
```

Comment:
Is there a scenario where we are unable to disable a cop for a multiline offense? If not, then I think it's fine to support only trailing comments.